src/sentry/sentry_metrics/multiprocess.py

```diff
@@ -0,0 +1,334 @@
+        tags = parsed_payload_value.get("tags", {})
+
+        new_tags = {mapping[k]: mapping[v] for k, v in tags.items()}
+
+        new_payload_value["tags"] = new_tags
+        new_payload_value["metric_id"] = mapping[metric_name]
+        new_payload_value["retention_days"] = 90
+
+        new_payload = KafkaPayload(
+            key=message.payload.key,
+            value=json.dumps(new_payload_value).encode(),
+            headers=message.payload.headers,
+        )
+        new_message = Message(
+            partition=message.partition,
+            offset=message.offset,
+            payload=new_payload,
+            timestamp=message.timestamp,
+        )
+        new_messages.append(new_message)
+
+    # TODO(meredith): Need to add kickin off the celery task in here, eventually
+    # we will use kafka to forward messages to the product data model
+    # messages = [
+    #     {"tags": m.payload["tags"], "name": m.payload["name"], "org_id": m.payload["org_id"]}
+    #     for m in self.__messages
+    # ]
+    # process_indexed_metrics.apply_async(kwargs={"messages": messages})
+
+    return new_messages
+
+
+class MetricsConsumerStrategyFactory(ProcessingStrategyFactory):
+    def __init__(
+        self,
+        process_messages: Callable[[Message[TPayload]], TProcessed],
+        next_step: Callable[
+            [Callable[[Mapping[Partition, Position]], None]], ProcessingStrategy[TProcessed]
+        ],
+        max_batch_size: int,
+        max_batch_time: float,
+        processes: int,
+        input_block_size: int,
+        output_block_size: int,
+        initialize_parallel_transform: Optional[Callable[[], None]] = None,
+    ):
+        self.__process_messages = process_messages
+        self.__next_step = next_step
+
+        self.__max_batch_time = max_batch_time
+        self.__max_batch_size = max_batch_size
+
+        self.__processes = processes
+
+        self.__input_block_size = input_block_size
+        self.__output_block_size = output_block_size
+
+        self.__initialize_parallel_transform = initialize_parallel_transform
+
+    def create(
+        self, commit: Callable[[Mapping[Partition, Position]], None]
+    ) -> ProcessingStrategy[TPayload]:
+        transform_function = self.__process_messages
+
+        parallel_strategy = ParallelTransformStep(
+            transform_function,
+            self.__next_step(commit),
+            self.__processes,
+            max_batch_size=self.__max_batch_size,
+            max_batch_time=self.__max_batch_time,
+            input_block_size=self.__input_block_size,
+            output_block_size=self.__output_block_size,
+            initializer=self.__initialize_parallel_transform,
+        )
+
+        strategy = BatchMessages(parallel_strategy, self.__max_batch_time, self.__max_batch_size)
+
+        return strategy
+
+
+def get_streaming_metrics_consumer(topic: str, **options: Dict[str, str]) -> StreamProcessor:
+    DEFAULT_BLOCK_SIZE = int(32 * 1e6)
```

Comment:
These should definitively be CLI parameters, we tend to tune them quite a bit in prod.